src/frontend/src/optimizer/plan_node/stream_eowc_over_window.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+
+use fixedbitset::FixedBitSet;
+use risingwave_common::util::sort_util::OrderType;
+use risingwave_pb::stream_plan::stream_node::PbNodeBody;
+
+use super::generic::{self, PlanWindowFunction};
+use super::utils::TableCatalogBuilder;
+use super::{ExprRewritable, PlanBase, PlanRef, PlanTreeNodeUnary, StreamNode};
+use crate::stream_fragmenter::BuildFragmentGraphState;
+use crate::TableCatalog;
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct StreamEowcOverWindow {
```

Comment:
Will we need to distinguish whether it's EOWC in fe?

Reply:
Wait, don't we already have over agg?

Reply:
Oh it's logical. I'm stupid

Reply:
> Will we need to distinguish whether it's EOWC in fe?

Yes we are. And @st1page is thinking maybe we need to distinguish EOWC/EOU in logical plan node, but still to be determined.